### PR TITLE
[frontend] add useOverviewLayoutCustomization hook (#6724)

### PR DESCRIPTION
**Pull request by Goumies**
### Proposed changes

* add useOverviewLayoutCustomization hook

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/6724

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+type OverviewLayoutCustomizationSettingsConfigurationWidth = HalfWidth | FullWidth;
+
+export interface OverviewLayoutCustomizationSettingsConfigurationParameters {
+  order: number;
+  width: OverviewLayoutCustomizationSettingsConfigurationWidth;
+}
+
+type OverviewLayoutCustomizationSettingsConfigurationWidgets = Map<string, OverviewLayoutCustomizationSettingsConfigurationParameters>;
+
+const threatActorIndividualConfiguration = new Map<string, OverviewLayoutCustomizationSettingsConfigurationParameters>();
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: Don't think you should use a Map as GraphQL won't be able to provide one, you should rely on a simple Object I think

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+defaultConfiguration.set('latestCreatedRelationships', { order: 5, width: 6 });
+defaultConfiguration.set('latestContainers', { order: 6, width: 6 });
+defaultConfiguration.set('externalReferences', { order: 7, width: 6 });
+defaultConfiguration.set('mostRecentHistory', { order: 8, width: 6 });
+defaultConfiguration.set('notes', { order: 9, width: 12 });
+
+const overviewLayoutCustomization = new Map<string, OverviewLayoutCustomizationSettingsConfigurationWidgets>();
+overviewLayoutCustomization.set(
+  'Threat-Actor-Individual',
+  threatActorIndividualConfiguration,
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: This should come from the useAuth hook.

File: opencti-platform/opencti-front/src/utils/tests/useOverviewLayoutCustomization.test.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    expectedThreatActorIndividualConfiguration.set('latestCreatedRelationships', { order: 5, width: 6 });
+    expectedThreatActorIndividualConfiguration.set('latestContainers', { order: 6, width: 6 });
+    expectedThreatActorIndividualConfiguration.set('externalReferences', { order: 7, width: 6 });
+    expectedThreatActorIndividualConfiguration.set('mostRecentHistory', { order: 8, width: 6 });
+    expectedThreatActorIndividualConfiguration.set('notes', { order: 9, width: 12 });
+
+    expect(
+      useOverviewLayoutCustomization(threatActorIndividualEntityType),
+    ).toEqual(
+      expectedThreatActorIndividualConfiguration,
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/tests/useOverviewLayoutCustomization.test.ts]: I don't see how this test will work in the future... I think you should test that keys exists, that width are set or stuff like that as it will be updatable

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 4]
```diff
@@ -155,61 +156,98 @@ const ThreatActorIndividualComponent = ({
...
-          <StixCoreObjectLatestHistory
-            stixCoreObjectId={threatActorIndividual.id}
-          />
-        </Grid>
+        {
+          Array.from(overviewLayoutCustomization.entries()).map(([key, { width }]) => {
+            switch (key) {
+              case 'details':
+                return (
+                  <Grid item={true} xs={width} style={{ paddingTop: 10 }}>
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: The grid should be outside the switch

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 5]
```diff
@@ -155,61 +156,98 @@ const ThreatActorIndividualComponent = ({
...
+                    <ThreatActorIndividualDetails
+                      threatActorIndividualData={threatActorIndividual}
+                    />
+                  </Grid>
+                );
+              case 'basicInformation':
+                return (
+                  <Grid item={true} xs={6} style={{ paddingTop: 10 }}>
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: Missing width

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 6]
```diff
@@ -155,61 +156,98 @@ const ThreatActorIndividualComponent = ({
...
+                return (
+                  <Grid item={true} xs={width} style={{ marginTop: 30 }}>
+                    <StixCoreObjectLatestHistory
+                      stixCoreObjectId={threatActorIndividual.id}
+                    />
+                  </Grid>
+                );
+              case 'notes':
+                return (
+                  <Grid item={true} xs={width} style={{ marginTop: -20 }}>
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: I don't understand all those different styles

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 4 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: The style can differ

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 6 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: The Notes component was not originally part of the Grid children.
When including it, the style broke.

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts (hunk 2 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: Ok

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts (hunk 1 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: The schema is exposed with an object with maps, in the front.
image

Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: I will do the same with the overview layout customization

Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: Passing an object would have complicated the access to a configuration based on the entity type sent to API.
=> 'Threat-Actor-Individual' can not be used as an object index. Whereas it can be used as is as a map key.

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 6 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: Plus, different style are set depending on the child component.
image

Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: I will investigate why the style was managed this way.

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts (hunk 2 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: Let me send this mock from the useAuth hook.

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 4 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: As we discussed, the Grid custom style is a global problem in the platform.
We will rebase the code on the branch dedicated to this refactor.

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 6 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: The investigation can be found in the brainstorm.
The code will be rebased on the Grid style refactor branch.

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql [hunk 7]
```diff
@@ -27,6 +27,15 @@ type DefaultValueAttribute {
   defaultValues: [DefaultValue!]!
 }
 
+type OverviewLayoutCustomizationParameter {
+  width: Int!
+}
+
+type OverviewLayoutCustomization {
+  key: String!
+  values: OverviewLayoutCustomizationParameter!
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql]: This should be width: Int! directly

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql [hunk 8]
```diff
@@ -45,6 +54,7 @@ type EntitySetting implements InternalObject & BasicObject {
   availableSettings: [String!]! @auth
   created_at: DateTime! @auth
   updated_at: DateTime! @auth
+  overviewLayoutCustomization: [OverviewLayoutCustomization!] @auth
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql:53]: Missing a ! after the []

File: opencti-platform/opencti-graphql/src/modules/threatActorIndividual/threatActorIndividual.ts [hunk 9]
```diff
@@ -82,6 +82,16 @@ const THREAT_ACTOR_INDIVIDUAL_DEFINITION: ModuleDefinition<StoreEntityThreatActo
       },
     },
   },
+  overviewLayoutCustomization: [
+    { key: 'details', values: { width: 6 } },
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/threatActorIndividual/threatActorIndividual.ts]: Should be width directly

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import useAuth from './useAuth';
+import { defaultConfiguration, OverviewLayoutCustomizationSettingsConfigurationParameters } from '../overviewLayoutCustomization';
+
+const { overviewLayoutCustomization } = useAuth();
+
+const useOverviewLayoutCustomization: (entityType: string) => Map<string, OverviewLayoutCustomizationSettingsConfigurationParameters> = (entityType) => {
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: Should be something like :
 ```
const useOverviewLayoutCustomization: (entityType: string) => [string, number][] = (entityType) => {
  const { overviewLayoutCustomization } = useAuth();
  if (!overviewLayoutCustomization || !overviewLayoutCustomization?.has(entityType)) {
    throw Error(`Not customizable overview for ${entityType}`);
  }
  return Array.from(overviewLayoutCustomization.get(entityType)?.entries());
```

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 11]
```diff
@@ -155,61 +157,99 @@ const ThreatActorIndividualComponent = ({
     threatActorIndividualFragment,
     data,
   );
+  const { overviewLayoutCustomization } = useAuth();
+  const threatActorIndividualOverviewLayoutCustomization = overviewLayoutCustomization?.get(threatActorIndividual.entity_type) ?? defaultConfiguration;
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: Should be useOverviewLayoutCustomization

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 12]
```diff
@@ -155,61 +157,99 @@ const ThreatActorIndividualComponent = ({
...
-            stixCoreObjectId={threatActorIndividual.id}
-          />
-        </Grid>
-        <Grid item={true} xs={6} style={{ marginTop: 30 }}>
-          <StixCoreObjectLatestHistory
-            stixCoreObjectId={threatActorIndividual.id}
-          />
-        </Grid>
+        {
+          Array.from(threatActorIndividualOverviewLayoutCustomization.entries()).map(([key, { width }]) => {
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: Should be threatActorIndividualOverviewLayoutCustomization.map(([key, width]) =>

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 13]
```diff
@@ -155,61 +157,99 @@ const ThreatActorIndividualComponent = ({
...
+            switch (key) {
+              case 'details':
+                return (
+                  <Grid key={'details'} item={true} xs={width} style={{ paddingTop: 10 }}>
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: Grid should be outside the switch

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx [hunk 14]
```diff
@@ -155,61 +157,99 @@ const ThreatActorIndividualComponent = ({
...
+                );
+              case 'basicInformation':
+                return (
+                  <Grid key={'basicInformation'} item={true} xs={width} style={{ paddingTop: 10 }}>
+                    <StixDomainObjectOverview stixDomainObject={threatActorIndividual} />
+                  </Grid>
+                );
+              case 'demographics-biographics':
+                return (
+                  <Fragment key={'parent-demographics-biographics'} >
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: Should be two parameters

File: opencti-platform/opencti-front/src/private/components/settings/sub_types/entity_setting/EntitySettingSettings.tsx [hunk 15]
```diff
@@ -38,6 +38,12 @@ export const entitySettingFragment = graphql`
         name
       }
     }
+    overviewLayoutCustomization {
+        key
+        values {
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/settings/sub_types/entity_setting/EntitySettingSettings.tsx]: Should be width directly

File: opencti-platform/opencti-front/src/private/Root.tsx [hunk 16]
```diff
@@ -302,6 +305,11 @@ const RootComponent: FunctionComponent<RootComponentProps> = ({ queryRef }) => {
   const bannerSettings = computeBannerSettings(settings);
   const platformModuleHelpers = platformModuleHelper(settings);
   const platformAnalyticsConfiguration = generateAnalyticsConfig(settings);
+  const overviewLayoutCustomizationEntries = entitySettingsData
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/Root.tsx:309]: I think you can make it simplier with new Map(Object.entries....)

Like :

```
const overviewNoMap = {
    'Threat-Actor-Individual': new Map(Object.entries(threatActorIndividualOverviewLayout)),
  };
  const overviewLayoutCustomization = new Map(Object.entries(overviewNoMap));

```

File: opencti-platform/opencti-front/src/private/Root.tsx [hunk 17]
```diff
@@ -277,6 +279,7 @@ const RootComponent: FunctionComponent<RootComponentProps> = ({ queryRef }) => {
   const settings = useFragment<RootSettings$key>(rootSettingsFragment, settingsFragment);
   const me = useFragment<RootMe_data$key>(meUserFragment, meFragment);
 
+  const entitySettingsData = entitySettings?.edges?.map((setting) => (useFragment<EntitySettingSettings_entitySetting$key>(entitySettingFragment, setting.node)));
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/Root.tsx:282]: As a start, may be we don't need fragment here to simplify

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql [hunk 18]
```diff
@@ -27,6 +27,15 @@ type DefaultValueAttribute {
   defaultValues: [DefaultValue!]!
 }
 
+type OverviewLayoutCustomizationParameter {
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql]: I thought we agreed yesterday to do only Front in first chunk ? What changed ?

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql (hunk 7 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql]: No.
I want to keep the object to destructure it. And, explicitly name the constant. No magic number.

File: opencti-platform/opencti-front/src/private/Root.tsx (hunk 16 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/Root.tsx:309]: Typing was not allowing this

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql (hunk 18 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql]: We agreed on passing the mocked config from the backend

File: opencti-platform/opencti-graphql/src/modules/threatActorIndividual/threatActorIndividual.ts (hunk 9 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/threatActorIndividual/threatActorIndividual.ts]: Let me disagree on this one.

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 13 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: This will be done after the merge of the Grid style refactoring.

File: opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts (hunk 10 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts]: I first attempted to call useAuth in the useOverviewLayoutCustomization hook but it was causing an error (illegal hook call).
We can pass the overviewLayoutCustomization to useOverviewLayoutCustomization.

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql (hunk 8 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql:53]: This must be nullable because the overview layout customization will be available on a subset of entity types.

File: opencti-platform/opencti-front/src/private/Root.tsx (hunk 17 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/Root.tsx:282]: What do you have in mind ?
I chose this like the previous lines above.

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/Root.tsx:282]: To get the data directly in the query

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql (hunk 7 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting.graphql]: I understand now.
Let me optimize the typing by changing the OverviewLayoutCustomizationParameter with the width property.

File: opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx (hunk 12 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/threats/threat_actors_individual/ThreatActorIndividual.tsx]: We are passing it in an object to ensure typing.
Otherwise, we would have to handle an array of strings and numbers.

File: opencti-platform/opencti-front/src/utils/hooks/useAuth.ts [hunk 19]
```diff
@@ -49,15 +51,24 @@ const defaultContext = {
...
 };
 export const UserContext = React.createContext<UserContextType>(defaultContext);
 
 const useAuth = () => {
-  const { me, settings, bannerSettings, entitySettings, platformModuleHelpers, schema } = useContext(UserContext);
+  const {
+    me,
+    settings,
+    bannerSettings,
+    entitySettings, // setting avec overview layout de chaque entité
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/hooks/useAuth.ts]: Can you remove comment ?

File: opencti-platform/opencti-front/src/utils/overviewLayoutCustomization.ts [hunk 20]
```diff
@@ -0,0 +1,11 @@
+export type OverviewLayoutCustomizationSettingsConfigurationWidgets = Map<string, number>;
+export type OverviewLayoutCustomization = Map<string, OverviewLayoutCustomizationSettingsConfigurationWidgets>;
+
+export const defaultConfiguration = new Map<string, number>();
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/overviewLayoutCustomization.ts]: As there is a default conf in modules now I think this file is no longer useful

Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/overviewLayoutCustomization.ts]: Where do we get the default configuration from when the feature flag is disabled, @Kedae?

Comment (Kedae) [reply opencti-platform/opencti-front/src/utils/overviewLayoutCustomization.ts]: Hm, yes, I have mistaken myself there. In fact the feature flag should be in the back and not in the front to enforce the default configuration in the back and note the one from elastic. My bad. Could you change that ?

File: opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+import { renderHook } from '@testing-library/react';
+import React from 'react';
+import useAuth from '../hooks/useAuth';
+import { createMockUserContext, ProvidersWrapper, ProvidersWrapperProps } from './test-render';
+
+describe('overviewLayoutCustomization', () => {
+  const threatActorIndividualEntityType = 'Threat-Actor-Individual';
+  const overviewLayoutCustomization = new Map();
+  const threatActorIndividualConfiguration = new Map();
+  threatActorIndividualConfiguration.set('details', { order: 1, width: 6 });
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx]: instead of doing a succession of 'set' you can directly write: newMap([[key1, value1], [key2, value2], ...]);

File: opencti-platform/opencti-front/src/utils/tests/test-render.tsx [hunk 22]
```diff
@@ -16,8 +16,24 @@ interface CreateUserContextOptions {
   entitySettings?: unknown,
   platformModuleHelpers?: unknown,
   schema?: unknown,
+  overviewLayoutCustomization?: unknown,
 }
 
+const fakeOverviewLayoutCustomization = new Map();
+const threatActorIndividualConfiguration = new Map();
+threatActorIndividualConfiguration.set('details', { order: 1, width: 6 });
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/tests/test-render.tsx]: same

File: opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+    ['details', 6],
+    ['basicInformation', 6],
+    ['demographics-biographics', 6],
+    ['latestCreatedRelationships', 6],
+    ['latestContainers', 6],
+    ['externalReferences', 6],
+    ['mostRecentHistory', 6],
+    ['notes', 12],
+  ]);
+  overviewLayoutCustomization.set(
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx]: same here, you can directly construct your map

File: opencti-platform/opencti-front/src/utils/tests/test-render.tsx [hunk 24]
```diff
@@ -16,8 +16,25 @@ interface CreateUserContextOptions {
...
+  ['details', 6],
+  ['basicInformation', 6],
+  ['demographics-biographics', 6],
+  ['latestCreatedRelationships', 6],
+  ['latestContainers', 6],
+  ['externalReferences', 6],
+  ['mostRecentHistory', 6],
+  ['notes', 12],
+]);
+fakeOverviewLayoutCustomization.set(
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/tests/test-render.tsx]: same

File: opencti-platform/opencti-front/src/utils/tests/test-render.tsx [hunk 25]
```diff
@@ -16,8 +16,25 @@ interface CreateUserContextOptions {
   entitySettings?: unknown,
   platformModuleHelpers?: unknown,
   schema?: unknown,
+  overviewLayoutCustomization?: unknown,
 }
 
+const fakeOverviewLayoutCustomization = new Map();
+const threatActorIndividualConfiguration = new Map([
+  ['details', 6],
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/tests/test-render.tsx:23]: why did you remove 'order' and 'width' ?
```suggestion
  ['details', { order: 1, width : 6 }],
```

Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/tests/test-render.tsx:23]: Because the model has evolved.
The order and width would be stored in the database but the configuration used for the template will be based on entries sorted by order with the value of the width.

Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/tests/test-render.tsx:23]: ok thanks

File: opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx (hunk 23 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx]: I can see why it was better to refactor the multiple set instructions.
For this last one, I do not understand. Why should this be constructed differently?

Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/tests/overviewLayoutCustomization.test.tsx]: It avoids the creation of an empty map some lines above, we directly create the map with its value. Also easier to read the constant value at its creation

**Review by Archidoit — APPROVED** [review]
